addons/mrp/models/stock_rule.py

```diff
@@ -33,15 +33,30 @@ def _onchange_action_operation(self):
 
     @api.model
     def _run_manufacture(self, procurements):
+        procurements_by_domain = defaultdict(list)
         productions_values_by_company = defaultdict(list)
         for procurement, rule in procurements:
             bom = self._get_matching_bom(procurement.product_id, procurement.company_id, procurement.values)
             if not bom:
                 msg = _('There is no Bill of Material of type manufacture or kit found for the product %s. Please define a Bill of Material for this product.') % (procurement.product_id.display_name,)
-
-            productions_values_by_company[procurement.company_id.id].append(rule._prepare_mo_vals(*procurement, bom))
                 raise ProcurementException(msg, [msg], [procurement])
 
+            procurements_by_domain[tuple(rule._get_domain_run_manufacture(bom, procurement))].append(procurement)
+
+        for domain, procurements in procurements_by_domain.items():
+            existing_production = self.env['mrp.production'].search(domain).filtered(lambda p:
+                all(m.bom_line_id and m.quantity_done == 0.0 for m in p.move_raw_ids)
+            )
```

Comment:
isn't it the same as looking for 'progress' state mo?

Reply:
also the commit message indicates only mo created by a RR will be merged but it's not true i think? all mo created by run will be affected?

also what happens with chaine moves/mto? if the finished move is linked to a move and then you "merge" another mo which should be linked to another move? my point is that you're not really merging here, you only increment a qty to produce